includes/base_controls/QControlBase.class.php

```diff
@@ -1635,7 +1635,16 @@ public function RenderWithName($blnDisplayOutput = true) {
 		 * @return string
 		 */
 		public function RenderComment($strType) {
-			return  QHtml::Comment( $strType . ' ' . get_class($this) . ' ' . $this->strName . ' id:' . $this->strControlId);
+			// We render the comment only if the debug mode is on
+			if (defined('__DEBUG_MODE__') && __DEBUG_MODE__ == true) {
+				if($this instanceof QBlockControl) {
```

Comment:
Two reasons against it:

1. We have just one place where we are rendering the comment this way. 
2. We are using the instanceof to print the template path. 

This one is concise, in my opinion.